modules/diskq/qdisk.c

```diff
@@ -560,18 +601,12 @@ _load_state(QDisk *self, GQueue *qout, GQueue *qbacklog, GQueue *qoverflow)
 
   if (!self->options->read_only)
     {
-      end_ofs = qout_ofs;
-      if (qbacklog_ofs && qbacklog_ofs < end_ofs)
-        end_ofs = qbacklog_ofs;
-      if (qoverflow_ofs && qoverflow_ofs < end_ofs)
-        end_ofs = qoverflow_ofs;
-      if(end_ofs > QDISK_RESERVED_SPACE)
-        _truncate_file(self, end_ofs);
+      end_ofs = qdisk_try_to_truncate_file_to_minimal(self);
     }
 
   if (!self->options->reliable)
     {
-      self->file_size = qout_ofs;
+      self->file_size = end_ofs;
```

Comment:
Shouldn't this be the maximum of (end_ofs and QDISK_RESERVED_SPACE)?

Reply:
Wouldn't it will work the same as currently, just implemented differently?

Reply:
Thanks for eplaining, I should fix this.

Reply:
Fixed.